uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/ContactWorkPatternTest.php

```diff
@@ -36,13 +37,31 @@ public function testThereCannotBeTwoWorkPatternsForTheSameEmployeeWithTheSameEff
         'pattern_id' => $workPattern2->id,
         'effective_date' => $effectiveDate,
       ]);
-    } catch(PEAR_Exception $e) {
-      $this->assertEquals('DB Error: already exists', $e->getMessage());
+    } catch(InvalidContactWorkPatternException $e) {
+      $this->assertEquals('This contact already have a Work Pattern with this effective date', $e->getMessage());
 
       return;
     }
 
-    $this->fail('Expected an DB error, but the contact work patternx was created successfully');
+    $this->fail('Expected an DB error, but the contact work pattern was created successfully');
+  }
+
+  public function testItCanBeUpdatedWithoutChangingTheContactIdAndTheEffectiveDate() {
+    $effectiveDate = CRM_Utils_Date::processDate('2016-01-01');
+
+    $contactWorkPattern = ContactWorkPattern::create([
+      'contact_id' => 2,
+      'pattern_id' => 1,
+      'effective_date' => $effectiveDate,
+    ]);
+
+    // Update the pattern_id, but contact_id and effective_date are still the same
+    ContactWorkPattern::create([
+      'id' => $contactWorkPattern->id,
+      'contact_id' => 2,
+      'pattern_id' => 2,
+      'effective_date' => $effectiveDate,
+    ]);
```

Comment:
We should probably do an assert here, asserting the id and the pattern_id to be sure the update went through.